### ISSUE TITLE
Specifications

### ISSUE THREAD

**Issue by alabaki** [simon-jouet/PyOPN#1]
Hello @simon-jouet I am currently working on an NodeJS project that requires reading from the OPN-2001 barcode scanner. May I ask where you found the specifications? They are not on Opticon's website.

Since I am not familiar with Python, I am unable to make much sense of the code.

Thanks!

**Comment by simon-jouet** [comment]
hi @alabaki,

I had to reverse engineer the desktop application and the packets sent from the barcode scanner, nothing official at all. I wrote that at the time because the desktop app was working only on windows and we wanted to run our own system on a raspberry pi.

The OPN-2001 USB is just a serial port so once you got a serial connection you just need to send and receive packets. If you look at the python code `class Frame` is the format of the frames and then some specific frame have been defined to get the content that has been scanned and to resync the time.

Can't help much more that i'm afraid that's an afternoon work from a few years ago so I don't remember much :)
